### DISCUSSION TITLE
how to make exe with pyinstaller

### DISCUSSION THREAD

**Discussion by Heryk13** [PaddlePaddle/PaddleOCR#11490] [Q&A] [answered]
## how to make exe from python file with paddleOCR

i am using auto-py-to-exe but i think it will work in pyinstaller too with the same parameters to collect data.

### install auto-py-to-exe
for installing auto-py-to-exe
`$ pip install auto-py-to-exe`
Then to run it
`$ auto-py-to-exe`

github page: https://github.com/brentvollebregt/auto-py-to-exe

### select you main file .py and add the paddle dependencies in the Advanced options

![auto-py-to-exe-configOptions](https://github.com/PaddlePaddle/PaddleOCR/assets/97933785/d168b1c1-7f8c-49df-bffb-20fa8a4df4f6)

it take me a while to discover this.
aq é brasil HUE HUE HUE HUE BR

#6875 #9951

**Comment by pikut** [comment]
In my case, script return "ModuleNotFoundError: No module named 'scipy.io'", but other problems gone - your solution fix it.

**Reply by Heryk13** [reply]
in that case if you add in the --hidden-import scipy i think it will fix too

**Reply by pikut** [reply]
Unfortunely does not work, the script return the same error. What kind of python version you use?

**Reply by Heryk13** [reply]
I am using 3.10.13, i tried with the 3.11 but i dont remember why it didn't worked

If in hidden import is not working, try inserting the scipy in collect-data or colect-all

**Reply by alejandroNoc** [reply]
For me it work with this , im trying to do with PPOCR after following and exploring finally i got it:
hope in somewhile this help to other persons

![image](https://github.com/user-attachments/assets/ad36a286-a888-4637-81c8-c8b260778f68)

**Reply by AkshrajKumar** [reply]
> in that case if you add in the --hidden-import scipy i think it will fix too

Sir please help me I am stuck on the executable file execution using pyinstaller for paddleocr. If you have 2 min than contact me to look through the project. My mail id:- ankitkumargola95484@gmail.com.

**Comment by Edmond-Lee-Zse-Wong** [comment]
I still find an error after using your method, maybe I should add more file?
![屏幕截图 2024-02-05 230908](https://github.com/PaddlePaddle/PaddleOCR/assets/56387202/a713331f-fc53-4b1b-9847-d6d79acbf83b)

**Reply by Heryk13** [reply]
i think if you add in the colect-data or collect-all scipy it may get work, or its because of python version, i am using 3.10.13. the above guy had the same error of you, but in my is working, if you could share some code i could test for you in my enviroment.

**Reply by pikut** [reply]
In last night I successful compile my Python Script to EXE file like @Heryk13 say! Probably You should add scipy to collect-data and then lmdb to collect-all (when You add scipy, next error will be with lmdb). I tested with 3.9.13 and 3.10.11 python versions.
![Zrzut ekranu 2024-02-08 174723](https://github.com/PaddlePaddle/PaddleOCR/assets/27273381/5e84491b-89a4-4830-b144-b4cced6c7736)

**Reply by Heryk13** [reply]
i'm glad to hear that.

**Reply by Edmond-Lee-Zse-Wong** [reply]
After adding scipy and lmdb, unfortunately it still doesn't work. Below are my codes, written in the simplest way possible.
```
import traceback
import time

try:
    from paddleocr import PaddleOCR
    ocr = PaddleOCR(use_angle_cls=True, lang="ch")  # need to run only once to download and load model into memory
    img_path = 'img5.png'
    result = ocr.ocr(img_path, cls=True)
    for idx in range(len(result)):
        res = result[idx]
        for line in res:
            print(line)
except Exception as e:
    traceback.print_exc()
    time.sleep(60)
```
After that is the error information
```
Traceback (most recent call last):
  File "test.py", line 7, in <module>
  File "PyInstaller\loader\pyimod02_importers.py", line 419, in exec_module
  File "paddleocr\__init__.py", line 14, in <module>
    from .paddleocr import *
  File "PyInstaller\loader\pyimod02_importers.py", line 419, in exec_module
  File "paddleocr\paddleocr.py", line 48, in <module>
    from tools.infer import predict_system
  File "E:\Work_Projects\Intelligence_Group\Overall_Venv\venv\Scripts\output\test\_internal\paddleocr\tools\infer\predict_system.py", line 33, in <module>
    import tools.infer.predict_det as predict_det
  File "E:\Work_Projects\Intelligence_Group\Overall_Venv\venv\Scripts\output\test\_internal\paddleocr\tools\infer\predict_det.py", line 31, in <module>
    from ppocr.data import create_operators, transform
  File "E:\Work_Projects\Intelligence_Group\Overall_Venv\venv\Scripts\output\test\_internal\paddleocr\ppocr\data\__init__.py", line 37, in <module>
    from ppocr.data.lmdb_dataset import LMDBDataSet, LMDBDataSetSR, LMDBDataSetTableMaster
  File "E:\Work_Projects\Intelligence_Group\Overall_Venv\venv\Scripts\output\test\_internal\paddleocr\ppocr\data\lmdb_dataset.py", line 17, in <module>
    import lmdb
ModuleNotFoundError: No module named 'lmdb'
```
I‘ve already added scipy and lmdb like you @pikut say
![屏幕截图 2024-02-15 222211](https://github.com/PaddlePaddle/PaddleOCR/assets/56387202/07f93fc7-3e9b-4b24-96be-dbfe45c50eb5)
I have no idea where the problem lies. /(ㄒoㄒ)/~~

- System Environment：win10
- Version：Paddle：2.5.2 PaddleOCR：2.7.0.3 Related components：python：3.10.1 pyinstaller：6.3.0

**Reply by Heryk13** [reply]
@Edmond-Lee-Zse-Wong is there any specific reason to use the python version 3.10.1? i am using anaconda, if you are going to use the python 3.10 i recommend using a more newer version of 3.10, maybe its the version  of python you are using. i will test tomorrow in a newer version of 3.10 and in the 3.10.1. recommend using anaconda too

**Reply by Edmond-Lee-Zse-Wong** [reply]
No, there is no specific reason to use 3.10.1. I'm just unsure which version I should choose. Previously I used 3.7.6, but Paddle doesn't support this version, so I switched to whichever version would make it runs. Thanks very much for your @Heryk13  passion and patience!

**Reply by Heryk13** [reply, marked as answer]
hey @Edmond-Lee-Zse-Wong i created a exe with your code. here is what i used

code:
`import traceback
import time

try:
    from paddleocr import PaddleOCR
    ocr = PaddleOCR(use_angle_cls=True, lang="ch")  # need to run only once to download and load model into memory
    img_path = "img.png"
    result = ocr.ocr(img_path, cls=True)
    for idx in range(len(result)):
        res = result[idx]
        for line in res:
            print(f"ocrText: {line[1][0]}")
except Exception as e:
    traceback.print_exc()
    time.sleep(60)
`

i tested in the python versions 3.10.13 and i saw the error scipy, and did what @pikut said.
like this.

![image](https://github.com/PaddlePaddle/PaddleOCR/assets/97933785/06e57cfb-5961-41e5-835b-91aac661c269)

and it worked i think you could try in the python version 3.10.13 or 3.9.13 and 3.10.11 that worked for @pikut too.

the requirements
[requirements.txt](https://github.com/PaddlePaddle/PaddleOCR/files/14314309/requirements.txt)

extra:
i tested with 3.11.7 too but it didn't worked because PyMuPDF, i tried this solutions but no luck too.
https://stackoverflow.com/questions/76379293/how-can-i-fix-the-error-in-pymupdf-when-installing-paddleocr-with-pip

**Reply by Edmond-Lee-Zse-Wong** [reply]
Finally it works!!! Thank you@Heryk13 @pikut  so much!!!

**Reply by Heryk13** [reply]
im glad to hear that 👍

**Reply by pikut** [reply]
That's cool!

**Reply by SubhadeepBanerjeeChowdhury** [reply]
Thank you so much, this worked!

**Comment by NguyenDucQuan12** [comment]
hi, i am using yolov8 along with paddleocr to read detected characters. I want to wrap it all up, I see you are also using paddleocr, ultralytics so wanted to ask for your opinion. When I repackage it, when I install it to another device (without any modules), it cannot run. I want all the modules to be repackaged and the user can just install the software. use. Can you give me your opinion? Thank you
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/68120446/af48e1ef-0ca7-4abc-a7bc-2048338830ab)
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/68120446/30754b17-9327-44d3-95bb-05ff73ab45c6)

**Reply by Heryk13** [reply]
Yes i am using yolov8 too and it worked without need to installing nothing in the user pc. Go on and try to package to exe if it gives some error come here and i will help

**Reply by NguyenDucQuan12** [reply]
thank you  so much, it's works. However when launching the application it shows some errors (this doesn't happen when I launch the program using a python file), is there any way to know what is happening in my software, have you tried? Have you ever encountered these situations? I am new to this field so I don't have much knowledge about it

**Reply by Heryk13** [reply]
@NguyenDucQuan12 yes there is a plenty of things you have to do to work in the exe. call me in my discord so i can help you.
heryk119

**Reply by NguyenDucQuan12** [reply]
My company doesn't allow external network connections, so I can't use discord. Can you show me some methods to work with exe, and find some bottlenecks in the code?

**Reply by Heryk13** [reply]
well its hard to tell what its worked and not worked, if you could share some code or some error you are seeing so i can test it will be more easy.

this is the code i was using to make it work. i cant send you the entire code because its very long, but this is the important part to make it work.


```
# you need this to load pt files from your directory of exe
if getattr(sys, "frozen", False):
    ROOT = sys._MEIPASS # type: ignore
else:
    ROOT = os.path.dirname(__file__)

def processFiles(self, fileName, fileContent):
        detection_model = AutoDetectionModel.from_pretrained(
            model_type='yolov8',
            model_path=os.path.join(ROOT, "model", "best.pt"), # here is loading the pt file from inside your exe
            device="cpu",  # or 'cuda:0'
        )
        ocrEngine = PaddleOCR(
            use_angle_cls=False,
            lang='en',
            show_log=False,
            use_gpu=False,
            rec_model_dir=os.path.join(ROOT, "model", "rec"),
            det_model_dir=os.path.join(ROOT, "model", "det"),
            cls_model_dir=os.path.join(ROOT, "model", "cls")
        )
```

**Reply by NguyenDucQuan12** [reply]
this is my code:
`# It took a long time for this run
license_plate_detect= YOLO("./model/best.pt")
ocr = PaddleOCR(use_angle_cls=True, lang='en', det=False, ret= False, show_log=False, save_crop_res=False
                , crop_res_save_dir='./output') # need to run only once to download and load model into memory

def get_license_plate(license_plate_crop):

    is_license_plate= False
    result_license_plate= ocr.ocr(license_plate_crop, cls=True)[0]
    
    license_plate_crop_cvt = Image.fromarray(license_plate_crop)
    
    if result_license_plate:
        license_plate = [line[1][0] for line in result_license_plate]

        license_plate = [i.upper() for i in license_plate]
        license_plate =''.join(license_plate) # bỏ các khoảng trắng
        is_license_plate, license_plate=license_complies_format(license_plate)
    else:
        license_plate='000000000'

    return is_license_plate, license_plate_crop_cvt, license_plate`
Every time I start the paddleocr model, it causes my software to create a similar software. I have to turn off the created software for it to run. My program has RAM increasing day by day, even though I'm not doing anything.
At some point, one of my two camera streams had to stop. I don't know why or how to fix it.
![main_window](https://github.com/PaddlePaddle/PaddleOCR/assets/68120446/cc6da987-32ed-4338-966d-26787cbf85b1)

**Reply by NguyenDucQuan12** [reply]
Is there any way I can download the paddleocr model and reuse it (like the yolo model), without having to download it to memory every time I launch the software?

**Reply by Heryk13** [reply]
oh i see, for what i understood i think its not only the yolo or paddleocr but something more like in the entire code and the tkinter you are using. i will need to see the entire code to help you more further and understand what you are trying to build if you want, is there some way you can contact me so we can do some meeting in google meet or whatever you prefer?


> Is there any way I can download the paddleocr model and reuse it (like the yolo model), without having to download it to memory every time I launch the software?

yes i recommend to do so, is what i am doing in the
```
rec_model_dir=os.path.join(ROOT, "model", "rec"),
det_model_dir=os.path.join(ROOT, "model", "det"),
cls_model_dir=os.path.join(ROOT, "model", "cls")
```

here is the link
https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.7/doc/doc_ch/models_list.md

**Reply by Heryk13** [reply]
here is what i used in my program.
[model.zip](https://github.com/PaddlePaddle/PaddleOCR/files/14645356/model.zip)

**Reply by NguyenDucQuan12** [reply]
I really want to receive help from you, but my English is not good.
`
  rec_model_dir=os.path.join(ROOT, "model", "rec"),
  det_model_dir=os.path.join(ROOT, "model", "det"),
  cls_model_dir=os.path.join(ROOT, "model", "cls")
`
When I download the model and unzip it, 3 files appear (inference.pdmodel, inference.pdiparams.info, inference.pdiparams). How do I use it exactly?

**Reply by NguyenDucQuan12** [reply]
I can share the code, and we can discuss it, is that okay with you?

**Reply by Heryk13** [reply]
my english is not soo good too lol, i am brazilian.

```
ocrEngine = PaddleOCR(
            use_angle_cls=False,
            lang='en',
            show_log=False,
            use_gpu=False,
            rec_model_dir=os.path.join(ROOT, "model", "rec"), # use in here
            det_model_dir=os.path.join(ROOT, "model", "det"), # use in here
            cls_model_dir=os.path.join(ROOT, "model", "cls") # use in here
        )
```
you just need to give the path of the 3 folders rec, det, cls to the PaddleOCR and the paddleOCR will automatically load the files inside

for me its okay to share the code, but it will be okay for your work? because in here its public so anyone can see your code, lets go to a more private location so i can help you without harming your privacy of your code

**Reply by NguyenDucQuan12** [reply]
It's okay, it's just a developing project, it can help newbies gain more experience, I'm happy to share with inexperienced people like me.

**Reply by Heryk13** [reply]
Okay i will be waiting

**Reply by NguyenDucQuan12** [reply]
Wait a minute, I'm trying to upload my code and models

**Reply by NguyenDucQuan12** [reply]
here my code:
https://github.com/NguyenDucQuan12/license_plate_app
We can discuss here

**Reply by NguyenDucQuan12** [reply]
I'll describe some problems that I haven't really found a solution for, when I open a window called "quick view", a new window opens and shows the camera (I want a window for the supervisor, 1 window for the driver) but when opening the window, 1 camera will freeze. When I create an app, the software seems to be slow, increasing RAM capacity in a very suspicious way, and I also want to find a way to debug when my software has become an exe. Thank you very much for giving me more interesting knowledge

**Reply by NguyenDucQuan12** [reply]
@Heryk13 I'm still waiting for you

**Reply by Heryk13** [reply]
Just a sec im, fixing somethings in my work so i can concentrate in your problem.

Almost done here

**Reply by NguyenDucQuan12** [reply]
It's okay, prioritize your work first

**Reply by Heryk13** [reply]
@NguyenDucQuan12 i need your best.pt file

used a file i have to test, i am facing others errors but i think i will need your best.pt file in the future to test some functions further.

`FileNotFoundError: [Errno 2] No such file or directory: 'model\\best.pt'`

got some
Không kết nối được với camera, đang kết nối lại trong 10s


please create a requirements.txt file next time
[requirements.txt](https://github.com/PaddlePaddle/PaddleOCR/files/14653362/requirements.txt)
```
Exception in Tkinter callback
Traceback (most recent call last):
  File "c:\Users\81908\miniconda3\envs\licencePlate\lib\tkinter\__init__.py", line 1921, in __call__
    return self.func(*args)
  File "c:\Users\81908\Documents\GitHub\license_plate_app\main.py", line 104, in callback
    App(tkinter.Tk(), "SmartVN", sources)
  File "c:\Users\81908\Documents\GitHub\license_plate_app\main.py", line 31, in __init__
    Cam1 = tkCamera(self.window, 0, 0, width, height, sources, text= "Cam toàn cảnh", source= sources[0][1])
  File "c:\Users\81908\Documents\GitHub\license_plate_app\tkCamera.py", line 20, in __init__
    self.vid = VideoCapture(self.source, width, height)
  File "c:\Users\81908\Documents\GitHub\license_plate_app\videocapture.py", line 30, in __init__
    raise ValueError(error_read_frame, video_source)
ValueError: ('Không thể kết nối tới camera có địa chỉ: ', 'rtsp://169.254.184.113:554/1/stream1/Profile1')

```

have any idea what it could be?
probally because i dont have any camera in my enviroment. how could we test this in some other way?
maybe using a public rtsp just to test your code.

**Reply by NguyenDucQuan12** [reply]
because my best file exceeds 25mb so github doesn't allow me to upload it, the other error that appears is that it says it can't connect to the camera, you can use 0 for your webcam, or replace it with the line some rtsp guide.
file best.pt:
https://drive.google.com/file/d/1y2ELqyCJ3V0j5vHxrVnptatq3BwbQTRB/view?usp=sharing

**Reply by NguyenDucQuan12** [reply]
I suggest you download this video, rename it demo, then you put it as image
https://www.youtube.com/watch?v=o4bRh9zzJaU
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/68120446/3c13df81-b769-4e05-85ae-3b08abb2cbef)

**Reply by Heryk13** [reply]
Okay i will try it

**Reply by NguyenDucQuan12** [reply]
@Heryk13  I have made some edits in the project, you can update it again for better testing

**Reply by Heryk13** [reply]
it worked, now lets start to resolve your problem.

**Reply by Heryk13** [reply]
can you put me in the colaborator so i can work with you?
i will create my branch separetelly

**Reply by Heryk13** [reply]
i will fork and do a pull request to your project lets, organize this

**Reply by NguyenDucQuan12** [reply]
I have added you to the group

**Reply by Heryk13** [reply]
thanks

**Reply by Heryk13** [reply]
hey i will create the json to make the executable(.exe) work

**Reply by NguyenDucQuan12** [reply]
okay, just do it

**Reply by Heryk13** [reply]
@NguyenDucQuan12 
hey i made a commit in my branch, with the exeConfig you can use the auto-py-to-exe to use that file and make the exe, just dont forget to change the add folders to your pc location because its in my pc path. the exe worked, i could see the demo video working. and made some changes in the script so the exe can work without problem

**Reply by NguyenDucQuan12** [reply]
thank you, let me try

**Reply by NguyenDucQuan12** [reply]
@Heryk13 How can you upload model files and videos, please guide me

**Reply by Heryk13** [reply]
do you mean the .pt file and video file?
i don't know exactly but i use the github desktop and the visual studio code, and just do the commit from visual studio code source control and it goes.
i have download
https://git-scm.com/downloads
https://desktop.github.com/
https://code.visualstudio.com/download

**Reply by NguyenDucQuan12** [reply]
@Heryk13 When I open the app it takes up a lot of memory, is there any way I can optimize it
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/68120446/6509f5a5-0a9c-479d-bd39-d573eb0c0b5e)
and when I press the button to open more windows, one of my two cameras freezes, it cannot open at the same time
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/68120446/4a0dbec6-fcbe-4a27-8f8e-76a77fd1047e)

**Reply by Heryk13** [reply]
yes its very laggy in performance, you need to show the images in multi-thread so it wont stop in the third window, and showing the images in multi-thread the program it self will not be so laggy.

**Reply by Heryk13** [reply]
in term of memory, its normal because you are using yolo and paddle, its 2 very big library that consume a lot of memory. the best you can do is do some process in multi-thread so the program it self will not be so laggy in performance

**Reply by NguyenDucQuan12** [reply]
I also opened it in multithreading, but there seems to be some mistake that I can't see. In the tkCamera.py file when calling the command in a new window:
```
def open_new_window(self, is_quick_view):
        self.new_window = threading.Thread(target= another_window, args=(self.vid,is_quick_view))
        self.new_window.start()
        return self.new_window
```

**Reply by Heryk13** [reply]
if you have time, you can change the tkinter to pyside, so you can use qt designer and wont need to hardcode the UI, and the pyside is more flexible to do something better than tkinter, tkinter is too simple.

**Reply by Heryk13** [reply]
i think you need to multi thread the video update itself, because i cant drag the window because the update of the video.

**Reply by NguyenDucQuan12** [reply]
I will update the UI using Qt, I also tried multithreading for video streaming from the camera, however it seems to be having some errors, it doesn't actually show or notify the problem so I can't solve it. I think it's in cv2

**Reply by NguyenDucQuan12** [reply]
@Heryk13  Do you have any suggestions for using multithreading in this case

**Reply by Heryk13** [reply]
if you are going to use the qt there is the threading of the qt you can use with is very good and flexible to a lot of things
Qthreads its the name of the thread in qt and you can use the qt designer wich can be very easy to draw UI.
https://build-system.fman.io/qt-designer-download

you will write the UI and export to py file after that with a command so you can use in your program, if you need any help text me.

**Reply by NguyenDucQuan12** [reply]
thank you, i will learn about qt

**Comment by shine-jayakumar** [comment]
> ## how to make exe from python file with paddleOCR
> i am using auto-py-to-exe but i think it will work in pyinstaller too with the same parameters to collect data.
> 
> ### install auto-py-to-exe
> for installing auto-py-to-exe `$ pip install auto-py-to-exe` Then to run it `$ auto-py-to-exe`
> 
> github page: https://github.com/brentvollebregt/auto-py-to-exe
> 
> ### select you main file .py and add the paddle dependencies in the Advanced options
> ![auto-py-to-exe-configOptions](https://private-user-images.githubusercontent.com/97933785/296555744-d168b1c1-7f8c-49df-bffb-20fa8a4df4f6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2OTAyNTcsIm5iZiI6MTcxNzY4OTk1NywicGF0aCI6Ii85NzkzMzc4NS8yOTY1NTU3NDQtZDE2OGIxYzEtN2Y4Yy00OWRmLWJmZmItMjBmYThhNGRmNGY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA2VDE2MDU1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4NTZhMzZiMWYyMjBmNjg4NzQ1ZGEzZmZlMDBkMGQ1ZGJlZDZjNDRiYTE4YTFhZjk3ZTY2ODY3ZWNjYmYwOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qJB5CSNKRzE-T9gYNqU89B_LmEz2YL4HIlXvQi2cs4E)
> 
> it take me a while to discover this. aq é brasil HUE HUE HUE HUE BR
> 
> #6875 #9951

Thank you @Heryk13. You solution worked perfectly for me for Python 3.11.
For Python 3.10, I had to edit Python310/Lib/dis.py as per [this stackoverflow answer](https://stackoverflow.com/questions/69776492/indexerror-tuple-index-out-of-range-when-i-try-to-create-an-executable-from-a-p)

Thanks a lot again.

**Reply by Heryk13** [reply]
im glad to hear that. good luck on your project

**Comment by wgsthink** [comment]
Thank you, your solution solved my problem.

**Reply by Heryk13** [reply]
😁😁😁 i'm glad to hear that

**Comment by tangyanglm** [comment]
Thank you, this method also solved my problem

**Comment by stephen617hong** [comment]
<img width="944" alt="微信图片_20240701165720" src="https://github.com/PaddlePaddle/PaddleOCR/assets/173433543/ae63de16-db47-4e0d-a5d7-d46746829427">
After installing your steps, this prompt appears

**Reply by Heryk13** [reply]
what python version are you using?
what are the libraries you are using?
what are you trying to do? make a build for exe or installing something?

searching about cublasLt64 maybe is related to something in tensorflow

**Reply by stephen617hong** [reply]
it's working,thanks.

**Comment by Suvoo** [comment]
This is a great solution, might miss some corner cases with dll files,
perfectly working solution (August 2024) :  https://github.com/PaddlePaddle/PaddleOCR/discussions/11342

Tl,DR : 
`pyinstaller main.py --collect-all paddleocr --collect-all pyclipper --collect-all imghdr --collect-all skimage --collect-all imgaug --collect-all scipy.io --collect-all lmdb`

**Comment by Maximory77** [comment]
I tried a build with Pyinstaller with your method and it worked. It created the build without any problem. Unfortunately, when I test the exe created with the build, the execution time is much, much slower. It is so slow that I don't have any output after 30 minutes. When I use my program on runtime python, I start to have ouputs in 10 seconds and the program finish in around 400 s.
For precision, here is the command I used to create my build :

`pyinstaller --noconfirm --onedir --console --hidden-import "torch" --hidden-import "pyclipper" --hidden-import "imghdr" --hidden-import "imgaug" --collect-data "paddle" --collect-all "paddleocr" --collect-data "scipy" --collect-all "lmbd" --collect-all "skimage" sliding_window.py`

If you have any idea why it is doing it. Please, let me know if you have an idea to fix my problem.

**Reply by Heryk13** [reply]
sorry no idea what it could be, maybe if you put some print in your code and make the exe, you could log what is happening inside and discover it out from console.

**Reply by Maximory77** [reply]
Ok now it is working for some reason. I changed my source code and this is how I initialised the ocr function :

`slice = {'horizontal_stride': 2000, 'vertical_stride': 1000, 'merge_x_thres': 100, 'merge_y_thres': 60}
    ocr = PaddleOCR(use_angle_cls=True, gpu_mem = 2000,
                    save_crop_res = False,
                    crop_res_save_dir = output_folder + "/crop",
                    benchmark = False,
                    save_log_path = output_folder,
                    show_log = True,
                    savefile = False,
                    onnx = False,
                    tensorrt=False,
                    lang='latin',
                    use_dilation = True,
                    det_limit_side_len = 2500,
                    max_batch_size = 10,
                    cls_batch_num = 6,
                    rec_batch_num = 6
                    )`

Before I used the same parameters excepts these : 
`onnx = True, tensorrt = True, gpu_mem = 4000, max_batch_size =20, cls_batch_num = 12, rec_batch_num =12`
One or more of these paramaters should be the source of the problem.

**Comment by timminator** [comment]
I've created a standalone executable version of PaddleOCR that behaves exactly like the command line version of PaddleOCR when it is installed for example via pip.
I've created a CPU and GPU version and also setup installers for both of them for an even easier usage.
If you are interested, you can find my project [here](https://github.com/timminator/PaddleOCR-Standalone).

**Comment by bigBoLang** [comment]
**about tqdm problem：**
``
 File "paddleocr\ppocr\utils\network.py", line 93, in _download
    with tqdm(total=(int(total_size) + 1023) // 1024) as pbar:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    with tqdm(total=(int(total_size) + 1023) // 1024, **disable=True**) as pbar:
